pkg/broker/handlers.go

```diff
@@ -41,7 +41,15 @@ func handleReadyz(w http.ResponseWriter, r *http.Request, _ httprouter.Params) {
 // handleReadCatalog advertises the classes of service we offer, and specifc plans to
 // implement those classes.
 func handleReadCatalog(w http.ResponseWriter, r *http.Request, _ httprouter.Params) {
-	JSONResponse(w, http.StatusOK, config.Config().Spec.Catalog)
+	catalogData, err := SnakeCaseCatalog(config.Config().Spec.Catalog)
```

Comment:
I think this would be prettier, and safer, if you just defined the correct types in `pkg/api/types.go` then in `pkg/apis/servicebroker/v1alpha1/conversion.go` e.g.:

```go
func (in ServiceCatalog) Convert() api.ServiceCatalog {
  out := api.ServiceCatalog{}
  out.Services = make([]api.ServiceOffering, len(in.Services)
  for i, o := range in.Services {
    out.Services[i] = o.Convert()
  }
  return out
}

func (in ServiceOffering) Convert() {
  // etc, etc ...
}
```

It's similar to how DeepCopy works in Kubernetes then.

Reply:
That said, good find, I must have just got lucky in testing in that the Service Catalog doesn't need any of the broken fields :sweat_smile:

Reply:
Hi,

good point if k8s is doing conversions like that. I was jumping back and forth between that and the current solution I have implemented - The idea to avoid additional structs had won...

Also `test/unit/util/types.go` already contains the correct structs. I will move that from tests to the actual code so it can be used to convert the structs.

FYI: CloudFoundry apparently also accepts some keys in camelCase (e.g. planUpdatable) - at least it provided the option to change plans despite that the default should be false if not set.

Reply:
Hah, you win points for finding those structure definitions and sharing code 😉

Yeah, we try to stick to the official spec 100%, interesting that CF does that, perhaps it has secret hidden Kubernetes integration...

Reply:
Hah, you win points for finding those structure definitions and sharing code 😉

Yeah, we try to stick to the official spec 100%, interesting that CF does that, perhaps it has secret hidden Kubernetes integration...